File: Homework4/Question3/src/Fish.java
public class Fish implements Swimmer {
    private String name;
    private Swimmer friend;

    public Fish(String name, Swimmer friend) {
        this.name = name;
        this.friend = friend;
    }

    @Override
    public String getName() {
        return name;
    }


    @Override
    public Swimmer getFriend() {
        return friend;
    }

    public static void testFish() {
        Jellyfish j = new Jellyfish("Andrew");
        Fish fish1 = new Fish("BJMan", j);
        Fish fish2 = new Fish("Max", fish1);
        System.out.println(j.getName() == "Andrew");
        System.out.println(j.getFriend() == j);
        System.out.println(fish1.getName() == "BJMan");
        System.out.println(fish1.getFriend() == j);
        System.out.println(fish2.getName() == "Max");
        System.out.println(fish2.getFriend() == fish1);
    }
}

File: Homework4/Fish.kt
class Fish : Swimmer {
    private val name: String
    private val friend: Swimmer

    constructor(name: String, friend: Swimmer) {
        this.name = name
        this.friend = friend
    }

    override fun getName(): String {
        return name
    }

    override fun getFriend(): Swimmer {
        return friend
    }

    companion object {
        fun testFish() {
            val j = Jellyfish("Andrew")
            val fish1 = Fish("BJMan", j)
            val fish2 = Fish("Max", fish1)
            println(j.getName() == "Andrew")
            println(j.getFriend() == j)
            println(fish1.getName() == "BJMan")
            println(fish1.getFriend() == j)
            println(fish2.getName() == "Max")
            println(fish2.getFriend() == fish1)
        }
    }
}